source/common/network/address_impl.h

```diff
@@ -291,6 +307,29 @@ class Ipv6Instance : public InstanceBase {
     bool isUnicastAddress() const override {
       return !isAnyAddress() && !IN6_IS_ADDR_MULTICAST(&ipv6_.address_.sin6_addr);
     }
+    bool isLinkLocalAddress() const override {
+      // Check if the address is in the link-local range: fe80::/10 or in the v4 mapped link-local
+      // range: [::ffff:169.254.0.0].
+      return IN6_IS_ADDR_LINKLOCAL(&ipv6_.address_.sin6_addr) ||
+             (IN6_IS_ADDR_V4MAPPED(&ipv6_.address_.sin6_addr) &&
+              (ipv6_.address_.sin6_addr.s6_addr[12] == 0xa9 &&
+               ipv6_.address_.sin6_addr.s6_addr[13] == 0xfe));
+    }
+    bool isTeredoAddress() const override {
+      // Teredo addresses have the prefix 2001::/32.
```

Comment:
It would be clearer if this said `2001:0000::/32` (even though they're equivalent) as that's written in the RFC https://datatracker.ietf.org/doc/html/rfc4380#section-2.6

Reply:
done